src/Service.Tests/CosmosTests/QueryTests.cs

```diff
@@ -472,6 +479,141 @@ ... p
+
+            // cache configuration
+            RuntimeConfig configuration = ConfigurationTests.InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName, new EntityCacheOptions() { Enabled = true, TtlSeconds = 5 });
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+            const string CUSTOM_SCHEMA = "custom-schema.gql";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+            File.WriteAllText(CUSTOM_SCHEMA, SCHEMA);
+
+            string id = _idList[0];
+            string name;
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string mutation = @"
+mutation ($id: ID!, $partitionKeyValue: String!, $item: UpdatePlanetInput!) {
+    updatePlanet (id: $id, _partitionKeyValue: $partitionKeyValue, item: $item) {
+        id
+        name
+     }
+}";
+
+                string graphQLQuery = @$"
+query {{
+    planet_by_pk (id: ""{id}"") {{
+        id
+        name
+    }}
+}}";
+                string queryName = "planet_by_pk";
+
+                // First query
+                JsonElement queryResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                            client,
+                                            server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                            query: graphQLQuery,
+                                            queryName: queryName
+                                            );
+                name = queryResponse.GetProperty("name").GetString();
+
+                var update = new
+                {
+                    id = id,
+                    name = "new_name",
+                };
+
+                // Mutated name property
+                JsonElement mutationResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                client,
+                                server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                query: mutation,
+                                queryName: "updatePlanet",
+                                variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } },
+                                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(),
+                                clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED
+                                );
+
+                // Asserting name is mutated
+                Assert.IsFalse(queryResponse.GetProperty("name").GetString().Equals(update.name), "Mutation didn't change the name successfully");
+
+                // Second query - the data returned is from the cache, not the mutated one
```

Comment:
Wouldn't you want to invalidate the cache if it gets mutated? So, shouldn't name in queryResponse2 be the new value?